android/src/main/kotlin/com/apptentive/apptentive_flutter/ApptentiveFlutterPlugin.kt

```diff
@@ -48,23 +47,48 @@ class ApptentiveFlutterPlugin : FlutterPlugin, MethodCallHandler, ActivityAware
   }
 
   // Set channel method handler to null when plugin is detached
-  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) { channel.setMethodCallHandler(null) }
+  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
+    channel.setMethodCallHandler(null)
+  }
 
   // When plugin is attached to an Activity, register Apptentive Activity Callback
-  override fun onAttachedToActivity(binding: ActivityPluginBinding) { activity = binding.activity }
+  override fun onAttachedToActivity(binding: ActivityPluginBinding) {
+    activity = binding.activity
+
+    if (isApptentiveRegistered) {
+      Apptentive.registerApptentiveActivityInfoCallback(activityInfo)
+    }
+  }
 
   // When re-attached to activity, set current activity context and re-register Apptentive Activity Callback
-  override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) { activity = binding.activity }
+  override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
+    activity = binding.activity
+    Apptentive.registerApptentiveActivityInfoCallback(activityInfo)
+
+    Log.d(LogTag("Flutter"), "register ApptentiveActivityInfoCallback on onReattachedToActivityForConfigChanges")
+  }
 
   // When detached from activity, set current activity context to null
-  override fun onDetachedFromActivity() { activity = null }
+  override fun onDetachedFromActivity() {
+    activity = null
+    Apptentive.unregisterApptentiveActivityInfoCallback()
+
+    Log.d(LogTag("Flutter"), "unregister ApptentiveActivityInfoCallback onDetachedFromActivity")
+  }
 
   // When detached from activity, set current activity context to null
-  override fun onDetachedFromActivityForConfigChanges() { activity = null }
+  override fun onDetachedFromActivityForConfigChanges() {
+    activity = null
+    Apptentive.unregisterApptentiveActivityInfoCallback()
+
+    Log.d(LogTag("Flutter"), "unregister ApptentiveActivityInfoCallback onDetachedFromActivity")
+  }
 
   // Delegate the method call to the proper method
   override fun onMethodCall(call: MethodCall, result: Result) {
-    if (checkIfActivityIsNull(result)) { return }
+    if (checkIfActivityIsNull(result)) return
+    else Apptentive.registerApptentiveActivityInfoCallback(activityInfo)
```

Comment:
Calling `registerApptentiveActivityInfoCallback` here is necessary. This ensures the current activity is registered with the SDK. For the case like if the app goes to background or MC/Survey shows up, the SDK looses the Flutter activity context and doesn't show any interaction after that.